safe_transaction_service/history/services/transaction_service.py

```diff
@@ -256,14 +262,15 @@ def get_all_tx_hashes(
                 "ethereum_tx_id", "execution_date", "created", "block", "safe_nonce"
             )
             .distinct()
+            .order_by("-execution_date")
         )
 
         # Tricky, we merge SafeTx hashes with EthereumTx hashes
         queryset = (
-            multisig_safe_tx_ids.union(erc20_tx_ids)
-            .union(erc721_tx_ids)
-            .union(internal_tx_ids)
-            .union(module_tx_ids)
+            multisig_safe_tx_ids.union(erc20_tx_ids, all=True)
+            .union(erc721_tx_ids, all=True)
+            .union(internal_tx_ids, all=True)
+            .union(module_tx_ids, all=True)
             .order_by("-execution_date", "-safe_nonce", "block", "-created")
```

Comment:
> Use UNION instead of UNION ALL. There are already checks for duplicated elements   

Maybe the PR description should be UNION ALL instead UNION?

Reply:
Done